synapse/handlers/message.py

```diff
@@ -1244,16 +1254,42 @@ async def create_new_client_event(
 
                     state_map_for_event[(data.event_type, data.state_key)] = state_id
 
-                context = await self.state.compute_event_context(
+                context = await self.state.calculate_context_info(
                     event,
                     state_ids_before_event=state_map_for_event,
-                    # TODO(faster_joins): check how MSC2716 works and whether we can have
-                    #   partial state here
-                    #   https://github.com/matrix-org/synapse/issues/13003
                     partial_state=False,
                 )
+
             else:
-                context = await self.state.compute_event_context(event)
+                entry = (
+                    await self.hs.get_state_handler().resolve_state_groups_for_events(
+                        event.room_id,
+                        event.prev_event_ids(),
+                        await_full_state=False,
+                    )
+                )
+
+                if entry.state_group is None:
```

Comment:
Would it make more sense to move this code into `calculate_context_info` which is doing the same thing anyway? And then we can simplify it to remove the ability to pass in the `_StateCacheEntry`?

Reply:
done